chunked_stream/test/read_chunked_stream_test.dart

```diff
@@ -24,4 +26,22 @@ void main() {
     })();
     expect(await readChunkedStream(s), equals(['a', 'b', 'c']));
   });
+
+  test('readByteStream', () async {
+    final s = (() async* {
+      yield [1, 2];
+      yield Uint8List.fromList([3]);
+      yield [4];
+    })();
+    expect(await readByteStream(s), equals([1, 2, 3, 4]));
+  });
+
+  test('readByteStream returns Uint8List', () async {
+    final s = (() async* {
```

Comment:
Do you want to check that this also happens when the input does not consist of Uint8Lists?